tasks/main.yml

```diff
@@ -23,3 +23,22 @@
   apt: update_cache=yes name=linux-generic-lts-vivid state=present
   when: ansible_kernel | version_compare("3.18.0", "lt")
   register: upgrade_kernel
+
+- name: upgrade kernel to latest version
+  apt: upgrade=dist update_cache=yes dpkg_options='force-confold,force-confdef'
+  when: ubuntu_dist_upgrade|bool
+
+- name: reboot instance
+  command: /sbin/shutdown -r now
+  register: reboot_result
+  when: ubuntu_dist_upgrade|bool
```

Comment:
We should make this only reboot if kernel did get upgraded.  Currently this will reboot on every run where the ubuntu_dist_upgrade is set to true